src/conn.rs

```diff
@@ -451,6 +463,12 @@ impl<'a, 'c> InProgress<'a, 'c> {
             // TODO: consider making vocabulary lookup lazy -- we won't need it much of the time.
         }
 
+        // run any commands that we've created along the way.
+        if let Some(ref observer_service) = self.observer_service {
+            let mut os = observer_service.lock().unwrap();
+            os.run();
```

Comment:
I'm still not happy with the idea of starting a new thread each time we commit a transaction.

Reply:
I've been working on a "command_queue" observer that runs in a separate thread and fires when it notices a command has been added to the queue but I haven't managed to get it working yet. Was going to do as a followup.